nni/algorithms/compression/v2/pytorch/utils/evaluator.py

```diff
@@ -0,0 +1,581 @@
+
+    def revert_loss(self):
+        assert isinstance(self.model, pl.LightningModule)
+        self.model.training_step = self._ori_model_attr['training_step']
+
+    def patch_optimizer_step(self, before_step_tasks: List[Callable], after_step_tasks: List[Callable]):
+        assert isinstance(self.model, pl.LightningModule)
+
+        class OptimizerCallback(Callback):
+            def on_before_optimizer_step(self, trainer: pl.Trainer, pl_module: pl.LightningModule, optimizer: Optimizer, opt_idx: int) -> None:
+                for task in before_step_tasks:
+                    task()
+
+            def on_before_zero_grad(self, trainer: pl.Trainer, pl_module: pl.LightningModule, optimizer: Optimizer) -> None:
+                for task in after_step_tasks:
+                    task()
+
+        old_configure_callbacks = self.model.configure_callbacks
+
+        def patched_configure_callbacks(_):
+            callbacks = old_configure_callbacks()
+            callbacks.append(OptimizerCallback())  # type: ignore
+            return callbacks
+
+        self.model.configure_callbacks = types.MethodType(patched_configure_callbacks, self.model)
+
+    def revert_optimizer_step(self):
+        assert isinstance(self.model, pl.LightningModule)
+        self.model.configure_callbacks = self._ori_model_attr['configure_callbacks']
+
+    def register_hooks(self, hooks: List[Hook]):
+        for hook in hooks:
+            hook.register()
+            self._hooks.append(hook)
+
+    def remove_all_hooks(self):
+        for hook in self._hooks:
+            hook.remove()
+        self._hooks.clear()
+
+    def train(self, max_steps: int | None = None, max_epochs: int | None = None):
+        assert isinstance(self.model, pl.LightningModule)
+        # reset trainer
+        self.trainer: pl.Trainer = self.trainer.trace_copy().get()  # type: ignore
+        # NOTE: lightning may dry run some steps at first for sanity check in Trainer.fit() by default,
+        # If we want to record some information in the forward hook, we may get some additional information,
+        # so using Trainer.num_sanity_val_steps = 0 disable sanity check.
+        self.trainer.num_sanity_val_steps = 0
+
+        ori_max_steps, ori_max_epochs = self.trainer.fit_loop.max_steps, self.trainer.fit_loop.max_epochs
+        if max_steps:
+            self.trainer.fit_loop.max_steps = max_steps
+        if max_epochs:
+            self.trainer.fit_loop.max_epochs = max_epochs
+        self.trainer.fit(self.model, self.data_module)
+        self.trainer.fit_loop.max_steps, self.trainer.fit_loop.max_epochs = ori_max_steps, ori_max_epochs
+
+    def finetune(self):
+        self.train()
+
+    def evaluate(self) -> Tuple[float | None, List[Dict[str, float]]]:
+        """
+        NNI will use metric with key ``NNI_METRIC`` for evaluating model, please make sure you have this key in your ``Trainer.test()`` returned metric dicts.
```

Comment:
`NNI_METRIC` -> `default`.

If there is only one metric, this is not necessary?